applications/plugins/CGALPlugin/CGALPlugin_test/MeshFromImageWithFeatures_test.cpp

```diff
@@ -0,0 +1,222 @@
+/******************************************************************************
+*       SOFA, Simulation Open-Framework Architecture, development version     *
+*                (c) 2006-2017 INRIA, USTL, UJF, CNRS, MGH                    *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+
+#include <SofaTest/Sofa_test.h>
+//#include <SofaBoundaryCondition/FixedConstraint.h>
+#include <sofa/defaulttype/VecTypes.h>
+#include <sofa/simulation/Simulation.h>
+#include <SofaSimulationGraph/DAGSimulation.h>
+#include <sofa/simulation/Node.h>
+#include <SofaBaseMechanics/MechanicalObject.h>
+#include <SofaBaseMechanics/UniformMass.h>
+#include <SceneCreator/SceneCreator.h>
+#include <SofaBoundaryCondition/ConstantForceField.h>
+#include <SofaSimulationCommon/SceneLoaderXML.h>
+
+#define CGALPLUGIN_MESHGENERATIONFROMIMAGE_CPP
+#include <CGALPlugin/MeshGenerationFromImage.h>
+#include <image/ImageTypes.h>
+
+
+namespace sofa{
+namespace {
+
+using namespace modeling;
+using core::objectmodel::New;
+using sofa::simulation::SceneLoaderXML;
+using sofa::core::ExecParams;
+
+
+//template<typename DataTypes>
```

Comment:
What is the purpose of all the commented code ? 
In general every commented code should be removed.